src/kOS.Safe/Execution/CPU.cs

```diff
@@ -15,15 +15,18 @@ namespace kOS.Safe.Execution
 {
     public class CPU : ICpu
     {
-        private enum Status
+        private enum Section
         {
-            Running = 1,
-            Waiting = 2
+            Main = 1,
+            Trigger = 2
         }
 
         private readonly IStack stack;
         private readonly VariableScope globalVariables;
-        private Status currentStatus;
+        private Section currentRunSection;
+        private List<YieldFinishedDetector> triggerYields;
+        private List<YieldFinishedDetector> mainYields;
```

Comment:
These two lists contain the list of Pending Yields that haven't stated they're finished yet.  When a yiedler says it's finished, the CPU will remove it from the list.  Only when the list is empty again will the CPU resume.  This takes the place of the notion of Status = running or Status = waiting.